fix: correctly type testcafe bound queries

Meemaw opened the pull request:
It should be possible to configure `timeout` on per query level. This is extremely important on projects where default selector timeout is very low, but some specific queries want to be performed with higher timeouts (case on a project I work on).


Also strongly types the functions, giving nice autocomplete for options:

<img width="790" alt="Screenshot 2020-09-14 at 14 05 24" src="https://user-images.githubusercontent.com/8524109/93083860-84641b80-f693-11ea-976a-e44e0c2e0c1e.png">

benmonro commented:
@Meemaw thank you for the PR. One question; curious if you tried using `.with` for this case?  https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/with.html

My concern is that we're merging the testcafe options into the Dom testing library options and that we could run into conflicts down the road..  Having said that, I do like the simplicity of this approach.

Meemaw commented:
To be honed, I haven't tried `.with` before, but it works 🎉 .

That being said, I think number of options is rather small and we explicitly split them behind the scenes (not to pass them to dom testing library). Also, as you said it looks a bit simpler.

I can also remove but keep the updated types that work better now (before it was `any[]` and there was no auto completion)

Meemaw commented:
Okay, I think I prefer if we keep the function signature matching the underlying DOM testing library function signature. I've removed the option, but kept the updated types that will provide auto completion.

Also I've added a test case for the `.with({timeout}` option -- It cant hurt.

benmonro commented:
Agreed! Thanks as always

benmonro commented:
@Meemaw would you mind adding an example of this to the docs site?
